_posts/2023-04-30-writing-code-with-llm-dont-forget-kani.md

```diff
@@ -0,0 +1,306 @@
+---
+layout: post
+title: "Writing Code with LLM? Don't Forget Kani"
+---
+
+Kani is a verification tool that can help you prove properties about your Rust code.
+To learn more about Kani, check out [the Kani tutorial](https://model-checking.github.io/kani/kani-tutorial.html) and our [previous blog posts](https://model-checking.github.io/kani-verifier-blog/).
+
+
+Recently, ChatGPT---OpenAI’s AI chatbot built on top of large language models(LLM)---has taken the world by storm.
+People use ChatGPT to write blog posts, do homework, and even write code!
+That’s right---LLMs like ChatGPT can be used to write code, since code is essentially just another kind of text.
+And with ChatGPT, all you need to do is provide a text prompt describing the program you want, and it will generate the code for you - along with an explanation of how it arrived at the solution.
+
+We were also eager to explore the possibilities of using LLMs for daily coding tasks, so we decided to put ChatGPT to the test.
+In this blog post, we'll share our experiences with using ChatGPT and Kani to solve example coding problems, and explore how these technology could revolutionize the way we approach software development.
+The experiment we did include the following three tasks.
+
+1. We write prompt and generate (unverified!) code with ChatGPT.
+2. We verify the generated code to see if they are correct with Kani.
```

Comment:
"verify" and "to see if they are correct" are redundant. Perhaps say "We use Kani to check if the generated code is correct".